pkg/util/files/lookup_binary.go

```diff
@@ -0,0 +1,60 @@
+package files
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+type LookupPathFn = func() (string, error)
+
+// LookupNextToCurrentExecutable looks for the binary next to the current binary
+// Example: if this function is executed by /usr/bin/kuma-dp, this function will lookup for binary 'x' in /usr/bin/x
+func LookupNextToCurrentExecutable(binary string) LookupPathFn {
+	return func() (string, error) {
+		ex, err := os.Executable()
+		if err != nil {
+			return "", err
+		}
+		return filepath.Dir(ex) + "/" + binary, nil
+	}
+}
+
+// LookupInCurrentDirectory looks for the binary in the current directory
+// Example: if this function is executed by /usr/bin/kuma-dp that was run in /home/kuma-dp, this function will lookup for binary 'x' in /home/kuma-dp/x
+func LookupInCurrentDirectory(binary string) LookupPathFn {
```

Comment:
Is this really a good practice? Maybe useful for dev, but it opens potential for security issues (e.g. overwrite a file in tmp and trick the system into executing a malicious envoy).